### ISSUE TITLE
Arg not selection error when right-clicking inside TEdit

### ISSUE THREAD

**Issue by pamoroso** [Interlisp/medley#1912]
**Describe the bug**

Right-clicking inside the TEdit window (not on the title bar) yields a break window with the error:

```
In ERROR:
ARG NOT SELECTION
NIL
```

**To Reproduce**

1. launch Medley
2. open TEdit
3. right-click inside the TEdit window

**Expected behavior**

The right-click menu pops up with no errors.

**Screenshots**

The break window:

![tedit-error1](https://github.com/user-attachments/assets/6888aa04-ac83-4945-8f82-b8b4569da8a6)

The full desktop:

![tedit-error2](https://github.com/user-attachments/assets/ec0a0f89-ab6d-4dcf-9586-520664a37a26)

**Context (please complete the following information):**

 - OS: Linux Mint Cinnamon
 - OS Version: 22
 - Host arch: x86_64 (System76 Merkaat)
 - Maiko version: N/A
 - IL:MAKESYSDATE: 25-Nov-2024 04:27:51

**Additional context**

Te context of the error:

```
MOUSE/5(debug)BTV
   MESS1 "ARG NOT SELECTION"
   MESS2 NIL
   NOBREAK NIL
ERROR
\LISPERROR
\DTEST.UFN
   FROM NIL
   TO NIL
\TEDIT.COPYSEL
   SI::*CLEANUP-FORMS* SI::RESETUNWIND
   TEXTOBJ {TEXTOBJ}#144,172000
   DS #<Output Display Stream/130,24200>
   X 276
   Y 182
   SOURCESEL {S:? 1-0-L  100/40720}
   SELOPERATION PENDINGDEL
   SELFN NIL
   OSELOP NORMAL
   OLDX 276
   OLDY 182
   PREG (0 0 554 308)
   OSEL {S:? 0-0-L _ 100/40754}
   EXTENDFLG NIL
SI::*UNWIND-PROTECT*
   PANE {WINDOW}#122,65664
   LISPXHIST NIL
   SI::*RESETFORMS* ((& NIL))
   RESETSTATE NIL
MODERN-ORIG-\TEDIT.BUTTONEVENTFN
   WINDOW {WINDOW}#122,65664
   ORIGFUNCTION MODERN-ORIG-\TEDIT.BUTTONEVENTFN
   ANYWHERE NIL
   TITLEPROPORTION NIL
   CORNERREGION (883 352 562 328)
   TOPMARGIN 12
   CORNER NIL
   ATTACHEDREGION NIL
MODERNWINDOW.BUTTONEVENTFN
   WINDOW {WINDOW}#122,65664
   ORIGFUNCTION MODERN-ORIG-\TEDIT.BUTTONEVENTFN
   ANYWHERE NIL
   TITLEPROPORTION NIL
   CORNERREGION (883 352 562 328)
   TOPMARGIN "Tedit Window"
MODERNWINDOW.BUTTONEVENTFN
   W {WINDOW}#122,65664
   STREAM NIL
\TEDIT.BUTTONEVENTFN
   W {WINDOW}#122,65664
   STREAM NIL
\TEDIT.BUTTONEVENTFN
   SI::*DUMMY-FOR-CATCH* T
   SI::*CATCH-RETURN-FROM* (&)
\PROTECTED.APPLYA0001
   FN \TEDIT.BUTTONEVENTFN
   WINDOW {WINDOW}#122,65664
   SI::NLSETQ-VALUE NIL
   *PROCEED-CASES* (&)
   SI::*NLSETQFLAG* NIL
\PROTECTED.APPLY
   \MHCOM \TEDIT.BUTTONEVENTFN
   \MHPROCESS #<Process TEdit/174,24000>
   \MHWINDOW {WINDOW}#122,65664
   \MOUSEBUSY T
WINDOW.MOUSE.HANDLER
   \OLDTTY #<Process TEdit/174,24000>
   \MOUSEBUSY NIL
\MOUSE.PROCESS
   *FORM* (\MOUSE.PROCESS)
   *ARGVAL* NIL
   *TAIL* NIL
   *FN* \MOUSE.PROCESS
\EVALFORM
   %#FORM# (\MOUSE.PROCESS)
   *CURRENT-PROCESS* #<Process OLDMOUSE/174,24410>
   HELPFLAG BREAK!
   \CURRENTDISPLAYLINE 0
   \#DISPLAYLINES 40
   \LINEBUF.OFD #<IO Stream on T/174,74500>
   *READTABLE* #<ReadTable INTERLISP/174,73714>
   \PRIMTERMTABLE {TERMTABLEP}#174,66740
   \PRIMTERMSA {CHARTABLE}#174,67000
   TtyDisplayStream 
#<Output Display Stream/130,24300>
   SI::*RESETFORMS* NIL
   \INTERRUPTABLE T
   \TTYWINDOW NIL
   READBUF NIL
   \TERM.OFD #<Output Stream on T/170,117100>
   *STANDARD-OUTPUT* #<Output Stream on T/170,117100>
   *STANDARD-INPUT* #<IO Stream on T/174,74500>
\MAKE.PROCESS0
T
```

The issue occurs also in Medley Online.

**Comment by rmkaplan** [comment]
First, the right-click system-window menu is not the  expected behavior for a right-button click inside the Tedit window, the menu is only expected when you click in the title bar.  When you click inside the Tedit window, it should extend either the current selection of the current type (character/line/paragraph), or if you click over at the right margin, it should give you the option of closing a split window.

Second, from the screen image, it looks like you tried right-clicking in an empty Tedit window, no text had been typed in.  I have not been able to reproduce that error.  Nothing happens when I right-click inside an empty window, the caret just keeps flashing at the upper left, as I would expect.

Is there something else going on?

**Comment by rmkaplan** [comment]
What version are you running in?  I see it failing in online and current release, but not in fifth round (and I have done a lot of work there specifically on the empty window).

So maybe this should be marked as fixed in 5th round.

**Comment by pamoroso** [comment]
I added this to the list of [issues round 5 resolves](https://github.com/Interlisp/medley/pull/1857#issuecomment-2529380916).

**Comment by pamoroso** [comment]
Resolved by #1857.
